website/docs/sdk-reference/elixir.md

```diff
@@ -0,0 +1,251 @@
+ConfigCat.force_refresh();
+value = ConfigCat.get_value("key", "my default value") # Returns "value from server"
+```
+
+### Custom cache behaviour with `cache:` option parameter.
+To be able to customize the caching layer behaviour you need to implement the following interface:
+```elixir
+defmodule ConfigCat.ConfigCache do
+  alias ConfigCat.Config
+
+  @type key :: String.t()
+  @type result :: {:ok, Config.t()} | {:error, :not_found}
+
+  @callback get(key) :: {:ok, Config.t()} | {:error, :not_found}
+  @callback set(key, Config.t()) :: :ok
+end
+
+```
+
+At the moment, this is how the current `InMemoryCache` looks like:
+```elixir
+defmodule ConfigCat.InMemoryCache do
+  use GenServer
+
+  alias ConfigCat.ConfigCache
+  @behaviour ConfigCache
+
+  @spec start_link(options()) :: GenServer.on_start()
+  def start_link(options) do
+    name =
+      options
+      |> Keyword.fetch!(:cache_key)
+      |> name_from_cache_key()
+
+    GenServer.start_link(__MODULE__, :empty, name: name)
+  end
+
+  @impl ConfigCache
+  def get(cache_key) do
+    GenServer.call(name_from_cache_key(cache_key), :get)
+  end
+
+  @impl ConfigCache
+  def set(cache_key, value) do
+    GenServer.call(name_from_cache_key(cache_key), {:set, value})
+  end
+
+  defp name_from_cache_key(cache_key) do
+    String.to_atom(cache_key)
+  end
+
+  @impl GenServer
+  def init(state) do
+    {:ok, state}
+  end
+
+  @impl GenServer
+  def handle_call(:get, _from, :empty = state) do
+    {:reply, {:error, :not_found}, state}
+  end
+
+  @impl GenServer
+  def handle_call(:get, _from, state) do
+    {:reply, {:ok, state}, state}
+  end
+
+  @impl GenServer
+  def handle_call({:set, value}, _from, _state) do
+    {:reply, :ok, value}
+  end
+end
+```
+
+## Logging
+In the *ConfigCat SDK*, we use the default Elixir's [Logger](https://hexdocs.pm/logger/Logger.html) so you can customise as you like.
+
+```bash
+[info]  Fetching configuration from ConfigCat
+[info]  ConfigCat configuration json fetch response code: 200 Cached: "xxxxx-xxx"
+[debug] Evaluating get_value('show_contact_phone_number').
+```
+
+## `get_all_keys()`
+You can query the keys from your configuration in the SDK with the `get_all_keys()` method.
+
+```elixir
+keys = ConfigCat.get_all_keys()
+```
+
+## Using ConfigCat behind a proxy
+Provide your own network credentials (username/password), and proxy server settings (proxy server/port) by passing the proxy details to the creator method.
+
+```elixir
+{ConfigCat, [
+    sdk_key: "YOUR SDK KEY",
+    http_proxy: "https://user@pass:yourproxy.com"
+]}
+```
+
+## Sample Applications
+- <a href="https://github.com/configcat/elixir-sdk/tree/master/samples/" target="_blank">Sample App</a>
+
+## Look under the hood
+* <a href="https://github.com/configcat/elixir-sdk" target="_blank">ConfigCat's Elixir SDK on GitHub</a>
+* <a href="https://hex.pm/packages/configcat" target="_blank">ConfigCat's Elixir SDK on Hex.pm</a>
```

Comment:
I'm not sure if this is the best place, but we should also link to the HexDocs once we have them published.  It might be best to have that link near the top though, and not part of the "under the hood" section.

Reply:
I've added the HexDocs link to the list to the list. Since this is based on the other SDKs format I will let @laliconfigcat decide what he wants to do with the location of the link. 

To be honest, I know that having HexDocs is pretty standard but I'm not sure if this is something they will be investing time or wanting to send users to it since they want to have it all centralised on their /docs page.